### PR TITLE
56 add clear data button

### PR DESCRIPTION
**Pull request by Dan-Quin**
Finally got this to work, should be good to go.

### CODE REVIEW

File: frontend/src/slices/gameDataSlice.ts [hunk 1]
```diff
@@ -34,6 +34,11 @@ export const gameDataSlice = createSlice({
       state.currencyAmount = action.payload.currencyAmount;
       state.numBits = action.payload.numBits;
       state.upgrades = action.payload.upgrades;
+    },
+    resetGameData: (state) => {
```
Comment (jvgusch) [inline frontend/src/slices/gameDataSlice.ts:38]: I see we are manually resetting the state back to initial conditions. I think I remember you saying that you couldn't use `initialState` for some reason. Could you explain why that didn't work?

File: frontend/src/game/scenes/MainMenu.ts [hunk 2]
```diff
@@ -1,44 +1,63 @@
...
+      })
+      .setOrigin(0.5)
+      .setDepth(100);
 
     this.nextButton = this.add.rectangle(screenCenterX, 100, 100, 50, 0);
     this.nextButton.setInteractive();
-    this.nextButton.on('pointerdown', this.nextButtonClicked.bind(this));
-
+    this.nextButton.on("pointerdown", this.nextButtonClicked.bind(this));
+    this.resetButton = this.add.rectangle(screenCenterX, 600, 100, 50, 0xff000f);
```
Comment (josephmaxwellhyland) [inline frontend/src/game/scenes/MainMenu.ts:45]: We may want this button to actually have some text at some point

File: frontend/src/slices/gameDataSlice.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export const gameDataSlice = createSlice({
       state.numBits += action.payload.additionalBits;
     },
     sellData: (state) => {
-      state.currencyAmount += state.numBits / 10.0;
+      state.currencyAmount += Number((state.numBits / 10.0).toFixed(1));
```
Comment (josephmaxwellhyland) [inline frontend/src/slices/gameDataSlice.ts:25]: I think this change is unrelated to the issue, you can remove it

File: frontend/src/slices/gameDataSlice.ts (hunk 1 above)
Comment (josephmaxwellhyland) [reply frontend/src/slices/gameDataSlice.ts:38]: Also curious about this - if there's a way to do like `state = initialState` or something, that would make it much more sustainable if we add additional game data that has to be reset.

**Review by josephmaxwellhyland — COMMENTED** [review]
Overall good and functional! But if there's a way to utilize the `initialState`, that would help us avoid issues down the line.

File: backend/controllers/gameDataController.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ export const saveGame = asyncHandler(async (req: any, res) => {
 
       if (existingData) {
         existingData.numBits = gameData.numBits;
-        existingData.currencyAmount = gameData.currencyAmount;
+        existingData.currencyAmount = Number(gameData.currencyAmount.toFixed(1));
```
[the remark this answers is not in the document]
Comment (Dan-Quin) [reply backend/controllers/gameDataController.ts:17]: This is something put in from the currency bug fix, I have no idea why its showing as different.

[the remark this answers is not in the document]
Comment (josephmaxwellhyland) [reply backend/controllers/gameDataController.ts:17]: I think this may actually just be vestigial from then, the actual fix was in the **App.tsx** file: https://github.com/hto-projects/autohack-idle/pull/49/files

So you don't actually need this!

File: frontend/src/slices/gameDataSlice.ts (hunk 1 above)
Comment (Dan-Quin) [reply frontend/src/slices/gameDataSlice.ts]: This was changed with the new arrow function solution.

**Review by josephmaxwellhyland — APPROVED** [review]
LGTM!
